include/seqan3/range/container/aligned_allocator.hpp

```diff
@@ -84,72 +90,108 @@ class aligned_allocator
-     * \throws Throws std::bad_alloc if allocation fails.
+    /*!\brief Allocates sufficiently large memory to hold `n` many elements of `value_type`.
+     *
+     * \param[in] n The number of elements for which to allocate the memory.
+     *
+     * \returns The pointer to the first block of allocated memory.
+     *
+     * \throws Throws std::bad_alloc if allocation fails, i.e. either the call to the throwing version of
+     *         operator new function throws or the requested memory exceeds the maximal number of elements to allocate.
+     *
+     * \details
+     *
+     * Allocates `n * sizeof(value_type)` bytes of uninitialized storage by calling
+     * [operator new](https://en.cppreference.com/w/cpp/memory/new/operator_new).
+     * If the given `alignment` is bigger than
+     * [__STDCPP_DEFAULT_NEW_ALIGNMENT__](https://en.cppreference.com/w/cpp/memory/new/align_val_t) the alignment
+     * aware operator new that takes as second argument the alignment as std::align_val_t. The allocation methods
+     * used by the global operator new functions is unspecified, e.g. std::malloc or std::aligned_alloc. Accordingly,
+     * the requirements on the alignment and the size of allocated bytes is implementation defined.
+     *
      * \sa https://en.cppreference.com/w/cpp/memory/allocator/allocate
+     *
+     * ### Thread safety
+     *
+     * Thread-safe.
+     *
+     * ### Exception
+     *
+     * Strong exception guarantee.
      */
     [[nodiscard]]
-    pointer allocate(size_type n)
+    pointer allocate(size_type const n) const
     {
         constexpr size_type max_size = std::numeric_limits<size_type>::max() / sizeof(value_type);
         if (n > max_size)
-            throw std::bad_alloc();
-
-        // NOTE: On macOS glibc does not implement aligned_alloc, so we need to fallback to posix_memalign instead.
-#if defined(__APPLE__) && (!defined(_GLIBCXX_HAVE_ALIGNED_ALLOC) && !defined(_ISOC11_SOURCE))
-        void * p{};
-        if (int res = posix_memalign(&p, alignment, n * sizeof(value_type)); res == 0 && p != nullptr)
-            return static_cast<pointer>(p);
-#else
-        // NOTE:
-        // Allocate size bytes of uninitialized storage whose alignment is
-        // specified by alignment. The size parameter must be an integral
-        // multiple of alignment.
-        //
-        // Passing a size which is not an integral multiple of alignment or an
-        // alignment which is not valid or not supported by the implementation
-        // causes the function to fail and return a null pointer (C11, as
-        // published, specified undefined behavior in this case, this was
-        // corrected by DR 460).
-        // https://en.cppreference.com/w/cpp/memory/c/aligned_alloc
-        if (auto p = static_cast<pointer>(std::aligned_alloc(alignment, n * sizeof(value_type))))
-            return p;
-#endif
-
-        throw std::bad_alloc();
+            throw std::bad_alloc{};
```

Comment:
codecov is complaining that this is not tested.

Can you test it with a ridiculous high allocation request? Like, let's say `std::numeric_limits<size_type>::max()` for value_type `uint64_t`.

Reply:
yes we should test it.